src/libcrun/container.c

```diff
@@ -1892,7 +1892,7 @@ container_init (void *args, char *notify_socket, int sync_socket, libcrun_error_
     {
       ret = entrypoint_args->exec_func (entrypoint_args->container, entrypoint_args->exec_func_arg, exec_path,
                                         def->process->args);
-      _exit (ret);
+      return ret;
```

Comment:
we are returning ret but the error is not set.

Reply:
Yes I think all the handlers need a refactor so they could set `err` https://github.com/containers/crun/pull/795#issuecomment-983278779 . I might be overthinking this though.

Reply:
but before we can "return ret;" we need to make sure `err` is set.

I think the safest for now is to `_exit(EXIT_FAILURE);`

The handler can directly call _exit, so there is no need to return

Reply:
I might be wrong i am just worried if we `_exit` from different scope we will miss calling `cleanup attributes` which were configured in `libcrun_run_linux_container`  but if its not the case then i guess this PR is not needed.

Reply:
if we exit the process, I think we don't have to worry about cleaning up :-)

Reply:
makes sense i have modified logic a bit in new commit so we dont end up with negative return values

Reply:
I still believe we either create an error here (like we do at lines 1901-1903) or we call exit.  If we return only `ret` it might cause issues in the future where we assume the error is also set.

Reply:
SGTM, Should i add something like but instead of errno it has a `ret` which is the `return` code from handler.
```diff
--- a/src/libcrun/container.c
+++ b/src/libcrun/container.c
@@ -1892,6 +1892,9 @@ container_init (void *args, char *notify_socket, int sync_socket, libcrun_error_
     {
       ret = entrypoint_args->exec_func (entrypoint_args->container, entrypoint_args->exec_func_arg, exec_path,
                                         def->process->args);
+      if (ret != 0)
+        return crun_make_error (err, ret, "exec container process failed with as handler `%s`", entrypoint_args->context->handler);
+
       return ret;
     }
```

Reply:
Until all handlers are re-factored to properly handle error then we could remove this line and expect `err` to be already set.

Reply:
yes please

Reply:
Updated.